cmd/horcrux/cmd/leader_election.go

```diff
@@ -20,6 +20,46 @@ import (
 
 func init() {
 	rootCmd.AddCommand(leaderElectionCmd)
+	rootCmd.AddCommand(getLeaderCmd)
+}
+
+func sanitizeAddress(address string) (string, error) {
+	u, err := url.Parse(address)
+	if err != nil {
+		return "", fmt.Errorf("error parsing peer URL: %w", err)
+	}
+
+	hostname, port, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		return "", fmt.Errorf("error splitting host port from parsed URL: %w", err)
+	}
+
+	if strings.Contains(hostname, ":") {
+		// IPv6 Addreses need to be wrapped in brackets
+		return fmt.Sprintf("[%s]:%s", hostname, port), nil
+	} else {
+		return fmt.Sprintf("%s:%s", hostname, port), nil
+	}
+}
+
+func leaderElectMultiAddress(cosignerConfig *CosignerConfig) string {
+	var grpcAddresses []string
+
+	// Append local host:port
+	localAddr, err := sanitizeAddress(cosignerConfig.P2PListen)
+	if err == nil {
```

Comment:
Is there any danger to swallowing these errors? We never log or return them.

Reply:
good call, errors shouldn't be ignored. updated.